test/test_common/test_time_system.h

```diff
@@ -15,6 +17,34 @@ class TestTimeSystem : public Event::TimeSystem {
 public:
   ~TestTimeSystem() override = default;
 
+  /**
+   * This class will use the real monotonic time regardless of the time system in use (real
+   * or simulated). This should only be used when time is needed for real timeouts that govern
+   * networking, etc. It should never be used for time that only advances explicitly for alarms.
+   */
+  class RealTimeBound {
+  public:
+    template <class D>
+    RealTimeBound(const D& duration)
+        : end_time_(std::chrono::steady_clock::now() + duration) // NO_CHECK_FORMAT(real_time)
```

Comment:
i'm guessing you were swayed to use this style by the convenience of not bothering to pass timeSystem() into the ctor here.

Regardless; this turned out really well. Thanks!